src/main/deltachat.js

```diff
@@ -445,6 +445,41 @@ class DeltaChatController extends EventEmitter {
     return selectedChat
   }
 
+  _getChatById (chatId) {
+    if (!chatId) return null
+    const chat = this._dc.getChat(chatId).toJson()
+    if (!chat) return null
+
+    if (chatId === C.DC_CHAT_ID_DEADDROP) {
+      const chat = this._dc.getChat(C.DC_CHAT_ID_DEADDROP)
+      return (chat && Object.assign(chat.toJson(), { isDeaddrop: true })) || null
+    }
+
+    var messageIds = this._dc.getChatMessages(chat.id, C.DC_GCM_ADDDAYMARKER, 0)
+    // This object is NOT created with object assign to promote consistency and to be easier to understand
+    return {
```

Comment:
There is a repetition of code here that you might want to abstract out. Check out the function where the chat list is fetched, you'll see chats getting created (also with the summary). Can we reuse that code here so the front end always can guarantee that a chat object has the same properties, generated by the same code?

Reply:
we can, but I'm not sure if we should at that point... If I were too choose I would completely rewrite deltachat-node to be more object oriented and also typed. I personally would wait with refactoring that code until we make a big refactoring/code cleanup...

Reply:
Also I'm not sure about this whole dead drop topic.

Reply:
I'm suggesting a small improvement on this pr to reuse some existing code, that shouldnt require refactoring much else. Could be a source of bugs in the future to have two different ways to generate chat objects.

Reply:
we now have 3 different sources (selectedChat, chatList, getChatById) of those objects, but if its no memory issue I can optimize it as you suggest.

Reply:
I'm concerned about the messages property

Reply:
like if there are many messages the chat list object will get very very big and because it gets send over ipc it might slow down the application

Reply:
So I'll improve this thing I set the state of this pr back to `wip` and added some points to the todo list.

Reply:
Do you agree to these todo points? @karissa @Jikstra

Reply:
I optimized it and renamed the symbol/var name in the ChatList to make clear that the object there isn't the complete chat object. I'll mark this as resolved.